llvm/test/MC/Disassembler/AMDGPU/gfx11_dasm_vop3_from_vop1.txt

```diff
@@ -45,49 +46,64 @@
 # GFX11: v_bfrev_b32_e64 v255, 0xaf123456        ; encoding: [0xff,0x00,0xb8,0xd5,0xff,0x00,0x00,0x00,0x56,0x34,0x12,0xaf]
 0xff,0x00,0xb8,0xd5,0xff,0x00,0x00,0x00,0x56,0x34,0x12,0xaf
 
-# GFX11: v_ceil_f16_e64 v5, v1                   ; encoding: [0x05,0x00,0xdc,0xd5,0x01,0x01,0x00,0x00]
+# GFX11-REAL16: v_ceil_f16_e64 v5.l, v1.l        ; encoding: [0x05,0x00,0xdc,0xd5,0x01,0x01,0x00,0x00]
+# GFX11-FAKE16: v_ceil_f16_e64 v5, v1            ; encoding: [0x05,0x00,0xdc,0xd5,0x01,0x01,0x00,0x00]
 0x05,0x00,0xdc,0xd5,0x01,0x01,0x00,0x00
 
-# GFX11: v_ceil_f16_e64 v5, v255                 ; encoding: [0x05,0x00,0xdc,0xd5,0xff,0x01,0x00,0x00]
+# GFX11-REAL16: v_ceil_f16_e64 v5.l, v255.l      ; encoding: [0x05,0x00,0xdc,0xd5,0xff,0x01,0x00,0x00]
+# GFX11-FAKE16: v_ceil_f16_e64 v5, v255          ; encoding: [0x05,0x00,0xdc,0xd5,0xff,0x01,0x00,0x00]
 0x05,0x00,0xdc,0xd5,0xff,0x01,0x00,0x00
 
-# GFX11: v_ceil_f16_e64 v5, s1                   ; encoding: [0x05,0x00,0xdc,0xd5,0x01,0x00,0x00,0x00]
+# GFX11-REAL16: v_ceil_f16_e64 v5.l, s1          ; encoding: [0x05,0x00,0xdc,0xd5,0x01,0x00,0x00,0x00]
+# GFX11-FAKE16: v_ceil_f16_e64 v5, s1            ; encoding: [0x05,0x00,0xdc,0xd5,0x01,0x00,0x00,0x00]
```

Comment:
Why do sgpr operands not have `.l`/`.h`? VOP3 opsel can be used with sgprs.

Reply:
No reason other than it's just not supported yet.

Reply:
I think they are either supported already in the MC layer, or will be when we finish upstreaming the current changeset. The syntax is different though. SGPRs use the opsel:[] syntax.

Reply:
> SGPRs use the opsel:[] syntax.

That's an odd choice, s0.l/s0.h would be much more readable and I don't see any downside to displaying sgprs and vgprs the same way.

Reply:
SP3 uses s0.l/s0.h syntax.

Reply:
> > SGPRs use the opsel:[] syntax.
> 
> That's an odd choice, s0.l/s0.h would be much more readable and I don't see any downside to displaying sgprs and vgprs the same way.

Yep, I agree the .l/.h syntax would be better. It has not been a priority since it would only affect asm. We do not have codegen support for 16 bit sgprs coming any time soon. If you want to do any patches toward .l/.h syntax it would be welcome.

Reply:
How would something like `v_add_nc_i16 v0.h, v1.h, s0.h` be represented after the switch? Do you mean `v_add_nc_i16 v0, v1, s0 op_sel:[1,1,1]` will be accepted as well?

Reply:
It would work like this ( this is how it works downstream)
 v_add_nc_i16 v0.l, v1.h, s0 op_sel:[1,1,0]
                                                                                                                                       
>v_add_nc_i16 v0.l, v1.h, s0 op_sel:[1,1,0] ; encoding: [0x00,0x18,0x0d,0xd7,0x01,0x01,0x00,0x00]
     
v_add_nc_i16 v0.l, v1.h, s0 op_sel:[1,0,0]
                                                              
>v_add_nc_i16 v0.l, v1.h, s0 op_sel:[1,0,0] ; encoding: [0x00,0x08,0x0d,0xd7,0x01,0x01,0x00,0x00]      

Suffixes on the vgprs are required. If you want to use a hi/lo sgpr, you must pass the op_sel:[] operand, and the bits in that operand corresponding to vgprs must match the suffixes on the vgprs (there is verification they match).

It is a bit awkward but not impossible I think to support suffixes on the sgprs. For vgprs there is an actual register number for hi vs lo, and when you parse src0 you set the register number. For SGPRs you would want to set the bit in src0_modifiers.

Reply:
I see, thanks. Awkward, but OK as a temporary solution.